paymentsheet/src/main/java/com/stripe/android/paymentsheet/elements/EmailConfig.kt

```diff
@@ -0,0 +1,50 @@
+package com.stripe.android.paymentsheet.elements
+
+import android.util.Patterns
+import androidx.compose.ui.text.input.KeyboardType
+import com.stripe.android.paymentsheet.elements.common.TextFieldConfigInterface
+import com.stripe.android.paymentsheet.elements.common.TextFieldElementState
+import com.stripe.android.paymentsheet.R
+import java.util.regex.Pattern
+
+internal class EmailConfig(private val pattern: Pattern = Patterns.EMAIL_ADDRESS) :
+    TextFieldConfigInterface {
+    override val debugLabel = "email"
+    override val label = R.string.becs_widget_email
+    override val keyboard = KeyboardType.Email
+
+    override fun filter(userTyped: String) =
+        userTyped.filter { Character.isLetterOrDigit(it) || it == '.' || it == '@' }
```

Comment:
I am going to just allow any characters to be inserted and just leave the determineState to determine it if is valid or not.  Especially since the keyboard is intended for email.   

I have it setup so that:
"" -> Blank and Required -> error not shown, params not generated
"sdf -> Incomplete -> error shown when does not have focus
"@." -> Incomplete -> error shown when does not have focus
"@.s" -> Malformed -> error shown when does have focus